src/utilities/firebase/firebaseActions.js

```diff
@@ -39,6 +39,39 @@ export const addSpecialDiet = (diet) => {
     });
 };
 
+export const addDairyOption = (items) => {
+  var currentU = authenticateUser();
+
+  firebase
+    .database()
+    .ref(`userID/${currentU}/${constants.preference}`)
+    .update({ [constants.preference]: items }, (error) => {
+      errorHandling(error);
+    });
+};
+
+export const addProduceOption = (items) => {
+  var currentU = authenticateUser();
+
+  firebase
+    .database()
+    .ref(`userID/${currentU}/${constants.preference}`)
+    .update({ [constants.produceOption]: items }, (error) => {
+      errorHandling(error);
+    });
+};
+
+export const addBakingOption = (items) => {
+  var currentU = authenticateUser();
+
+  firebase
+    .database()
+    .ref(`userID/${currentU}/${constants.preference}`)
+    .update({ [constants.bakingOption]: items }, (error) => {
+      errorHandling(error);
+    });
+};
```

Comment:
same here

Reply:
tbh I dont even think the API supports these ingredients, maybe not worth changing for now. I mostly left it so we could refactor these heading (baking, dairy.. ) maybe down the line to something else since they seem to be distinct.  Let me know if you think I should actually consolidate it

Reply:
well for simplicity sakes making it into 1 node seems easier. Rather than having to manage 3 functions that return ingredients, you have 1 function that returns the equivalent of 3 functions. Less calls also improve performance and cost

Reply:
I would make it under 1 node